src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            int salaryInfo = 0;
+            for (String employeeData : data) {
+                String[] splitData = employeeData.split(" ");
+                String date = splitData[0];
+                String employeeName = splitData[1];
+                int hours = Integer.parseInt(splitData[2]);
+                int moneyForHour = Integer.parseInt(splitData[3]);
+                LocalDate localDate = LocalDate.of(getYear(date), getMonth(date), getDay(date));
+                LocalDate fromDate = LocalDate.of(getYear(dateFrom),
+                        getMonth(dateFrom), getDay(dateFrom));
```

Comment:
this variable doesn't change, create them before loop instead of recreating each time